pkg/chaosdaemon/tasks/pod_handler.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tasks
+
+import (
+	"sync"
+
+	"github.com/go-logr/logr"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/cerr"
+)
+
+var ErrNotPodPID = cerr.NotType[PodID]()
+
+var ErrPodProcessMapNotInit = cerr.NotInit[map[PodID]SysPID]().WrapName("PodProcessMap").Err()
+
+type PodID string
+
+func (p PodID) ToID() string {
+	return string(p)
+}
+
+// ChaosOnPOD stand for the inner process injector for container.
+type ChaosOnPOD interface {
+	Injectable
+	Recoverable
+}
+
+type PodProcessMap struct {
+	m      map[PodID]SysPID
+	rwLock sync.RWMutex
+}
+
+func NewPodProcessMap() PodProcessMap {
+	return PodProcessMap{
+		m:      make(map[PodID]SysPID),
+		rwLock: sync.RWMutex{},
+	}
+}
+
+func (p *PodProcessMap) Read(podPID PodID) (SysPID, error) {
+	p.rwLock.RLock()
+	defer p.rwLock.RUnlock()
+	sysPID, ok := p.m[podPID]
+	if !ok {
+		return SysPID(0), ErrNotFoundSysID.WithStack().Err()
+	}
+	return sysPID, nil
+}
+
+func (p *PodProcessMap) Write(podPID PodID, sysPID SysPID) {
+	p.rwLock.Lock()
+	defer p.rwLock.Unlock()
+	p.m[podPID] = sysPID
```

Comment:
Will this `podPID` be recycled? or is it a memory leak?

Reply:
I agree with it , it is a mem leak .

Reply:
I fixed it.